gr-blocks/lib/tag_debug_impl.cc

```diff
@@ -109,6 +123,11 @@ int tag_debug_impl::work(int noutput_items,
                 sout << tag.value << std::endl;
             }
         }
+
+        if (d_save_all) {
+            for (const auto& tag : d_work_tags)
+                d_tags.push_back(tag);
```

Comment:
Same here.

Reply:
I believe the `gr::thread::scoped_lock l(d_mutex);` in the beginning of the `work` function suffices.